src/ui/pages/Backup/useBackupContext.ts

```diff
@@ -0,0 +1,34 @@
+import { invariant } from 'src/shared/invariant';
+import { useSearchParams } from 'react-router-dom';
+import { windowContext } from 'src/ui/shared/WindowContext';
+
+export type BackupContext =
+  | {
+      appMode: 'onboarding';
+    }
+  | {
+      appMode: 'wallet';
+      groupId: string;
+    };
+
+/**
+ * Combines URL search parameters from the window's location and react-router's search params.
+ * @returns {URLSearchParams} Merged search parameters from both sources.
+ */
+function useUrlParams(): URLSearchParams {
+  const params = new URLSearchParams(window.location.search);
+  const [hashParams] = useSearchParams();
+  return new URLSearchParams([...params.entries(), ...hashParams.entries()]);
+}
+
+export function useBackupContext(): BackupContext {
+  const params = useUrlParams();
+
+  if (windowContext.isOnboardingMode()) {
+    return { appMode: 'onboarding' };
+  } else {
+    const groupId = params.get('groupId');
```

Comment:
Can groupId be inside global search params?
Do we really need to combine all params to get this value here?

Reply:
If we pass group id as query param, may be it is better to pass it as hash param from the first place?

Reply:
I removed `useBackupContext`

Reply:
Updated, not relevant anymore